commit 4009ddc69225d571772aaea597615a0c032e14ab
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Thu Sep 28 00:19:18 2017 +0200

    hurd: Fix `getifaddrs' and `freeifaddrs' symbol exposition
    
    from `freeaddrinfo'.
    
    `getifaddrs' and `freeifaddrs' are not in POSIX, they should not be
    exposed along `freeaddrinfo' (through `__check_pf') which is POSIX.
    
            * include/ifaddrs.h (__getifaddrs, __freeifaddrs): New declarations,
            and use libc_hidden_def on them.
            * inet/ifaddrs.c (__getifaddrs, __freeifaddrs): Use libc_hidden_def on
            them.
            * sysdeps/gnu/ifaddrs.c (__getifaddrs, __freeifaddrs): Likewise.
            * inet/check_pf.c (__check_pf): Use __getifaddrs and __freeifaddrs
            instead of getifaddrs and freeifaddrs.

Index: glibc-2.26/include/ifaddrs.h
===================================================================
--- glibc-2.26.orig/include/ifaddrs.h
+++ glibc-2.26/include/ifaddrs.h
@@ -9,6 +9,11 @@
 libc_hidden_proto (getifaddrs)
 libc_hidden_proto (freeifaddrs)
 
+extern int __getifaddrs (struct ifaddrs **__ifap);
+libc_hidden_proto (__getifaddrs)
+extern void __freeifaddrs (struct ifaddrs *__ifa);
+libc_hidden_proto (__freeifaddrs)
+
 struct in6addrinfo
 {
   enum {
Index: glibc-2.26/inet/check_pf.c
===================================================================
--- glibc-2.26.orig/inet/check_pf.c
+++ glibc-2.26/inet/check_pf.c
@@ -32,7 +32,7 @@ __check_pf (bool *seen_ipv4, bool *seen_
 
   /* Get the interface list via getifaddrs.  */
   struct ifaddrs *ifa = NULL;
-  if (getifaddrs (&ifa) != 0)
+  if (__getifaddrs (&ifa) != 0)
     {
       /* We cannot determine what interfaces are available.  Be
 	 pessimistic.  */
@@ -51,7 +51,7 @@ __check_pf (bool *seen_ipv4, bool *seen_
     else if (runp->ifa_addr->sa_family == PF_INET6)
       *seen_ipv6 = true;
 
-  (void) freeifaddrs (ifa);
+  (void) __freeifaddrs (ifa);
 }
 
 
Index: glibc-2.26/inet/ifaddrs.c
===================================================================
--- glibc-2.26.orig/inet/ifaddrs.c
+++ glibc-2.26/inet/ifaddrs.c
@@ -30,6 +30,7 @@ __getifaddrs (struct ifaddrs **ifap)
   return -1;
 }
 weak_alias (__getifaddrs, getifaddrs)
+libc_hidden_def (__getifaddrs)
 libc_hidden_weak (getifaddrs)
 stub_warning (getifaddrs)
 
@@ -43,5 +44,6 @@ __freeifaddrs (struct ifaddrs *ifa)
   abort ();
 }
 weak_alias (__freeifaddrs, freeifaddrs)
+libc_hidden_def (__freeifaddrs)
 libc_hidden_weak (freeifaddrs)
 stub_warning (freeifaddrs)
Index: glibc-2.26/sysdeps/gnu/ifaddrs.c
===================================================================
--- glibc-2.26.orig/sysdeps/gnu/ifaddrs.c
+++ glibc-2.26/sysdeps/gnu/ifaddrs.c
@@ -151,6 +151,7 @@ __getifaddrs (struct ifaddrs **ifap)
   return 0;
 }
 weak_alias (__getifaddrs, getifaddrs)
+libc_hidden_def (__getifaddrs)
 #ifndef getifaddrs
 libc_hidden_weak (getifaddrs)
 #endif
@@ -161,4 +162,5 @@ __freeifaddrs (struct ifaddrs *ifa)
   free (ifa);
 }
 weak_alias (__freeifaddrs, freeifaddrs)
+libc_hidden_def (__freeifaddrs)
 libc_hidden_weak (freeifaddrs)
